operator/src/main/java/org/projectnessie/operator/resource/NessieAwsSecret.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.operator.resource;
+
+public class NessieAwsSecret {
+
+  private String secretName = "awscreds";
+  private String awsAccessKeyId = "aws_access_key_id";
+  private String awsSecretAccessKey = "aws_secret_access_key";
```

Comment:
Can we support EC2 metadata-based access (when we're running in EC2, of course).

Reply:
Certainly (I don't know what metadata-based access is). For now I'm trying to achieve feature parity with the Helm chart.

Reply:
I'm a bit fuzzy on this myself, but I imagine it's some flag on the client side and the AWS client will figure out the rest. I believe actual authN/Z checks are based on the source VM identity, which we do not control anyway.